### ISSUE TITLE
Worktable not working properly when GT Tools are used in crafting

### ISSUE THREAD

**Issue by JasonMcRay** [ForestryMC/ForestryMC#750]
Worktable crafting does not work properly with GregTech
If I layout any recipe involving GT tool it crafts the item only once, then I need to re-layout the recipe for it toi work again. GT Tools takes damage when used in crafting, and their damage value is stored in NBT, which might be an issue.

**Reproduce steps:**
1) Layout a recipe for GT pickaxe head
![2015-07-06_15 55 03](https://cloud.githubusercontent.com/assets/4236895/8523801/a69e278c-23f7-11e5-8896-c9a3b96b0914.png)
2) Put required items in storage place of the Worktable
3) Pull out the pickaxe head from the output slot
4) Try to pull it out again, it doesnt work.

**Comment by mezz** [comment]
Thanks for the report!
I think you're right in making this a separate issue. I'll take a look soon.

**Comment by pyure** [comment]
mezz, if it helps, I'm pretty certain this was working fine just a few weeks ago.  I didn't begin noticing the problem until a recent Forestry update to a pack.

**Comment by mezz** [comment]
This should be fixed in the latest unstable build. I tested with IC2 hammers since Gregtech is much harder to get set up in a dev environment. Please let me know if you find any issues.

**Comment by JasonMcRay** [comment]
@mezz has this been pushed out in the Stable build (3.6.5.23) or is it still only in unstable builds?

**Comment by mezz** [comment]
@JasonMcRay This is not in stable yet. It probably should be now, but it was a big enough change that I wanted to leave it in unstable for a while to see if any new bugs popped up.

**Comment by mezz** [comment]
Just uploaded Stable 3.6.6 with this fix to Curse.

**Comment by JasonMcRay** [comment]
@mezz Thanks, unfortunately its still not working. it works fine if the tool is storing the damage in metadata (like IC2 tools) But GregTech is using NBT values to store its damage.

Pulled out from minetweaker:
<gregtech:gt.metatool.01:12>.withTag({ench: [{lvl: 2 as short, id: 35 as short}, {lvl: 2 as short, id: 21 as short}], "GT.ToolStats": {PrimaryMaterial: "Midasium", MaxDamage: 6400 as long, SecondaryMaterial: "TungstenSteel", Damage: 800 as long}}) - This is example of a GT Hammer. As you can see it stores its maximum damage as MaxDamage (long) and actual damage taken as Damage (long).

It is done this way because GT tools are working slightly differently than regular tools. They do not take 1 damage per action, they take damage depending on what they are used for.

**Comment by mezz** [comment]
Ah, I only tested IC2 tools because I assumed they were the same.

**Comment by pyure** [comment]
I think once upon a time they were.

Thx for persistence mezz, we really appreciate it hugely :)

**Comment by mezz** [comment]
I'm having trouble getting GT to work in my dev environment.
I get this useless and obnoxious crash report:

```
The GregTech-Addon has a Problem.
IT'S NOT MY FAULT!!! Below is how to fix it.
Seems another Mod added a Crafting Recipe with null Output. Tell the Developer of said Mod to fix that.
DO NOT COME TO ME WITH THIS CRASH. YOU CAUSED IT YOURSELF, AND I TOLD YOU HOW TO FIX IT!!!
```

http://pastebin.com/efQKxmLc

I checked my own recipes but none are obviously broken. 
It would be a useful error if it printed the recipe and the crafting items it tried.
As it is, it just makes me want to avoid GT.

**Comment by JasonMcRay** [comment]
I had same problem with the FT Unstable. I will try to help to find because of which recipe is GT Complaining. Also this issue is happening in GTu5.09 (i see you tested GT6.00, which is in early alpha).
Link to GT5: http://forum.industrial-craft.net/index.php?page=Thread&threadID=11488

**Comment by mezz** [comment]
Thanks. 

Taking a close look, it's crashing on Forestry's planks to slabs recipe.
I'm not sure why, but it's trying to see what the recipe crafts, but it fails because the items it gives have no NBT. 
I'll look more into what GT's trying to do so I can fix the issue.

**Comment by mezz** [comment]
Solved the recipe crash, now to fix the worktable

**Comment by mezz** [comment]
![2015-09-24_01 00 00](https://cloud.githubusercontent.com/assets/916092/10068322/68c08ea6-6258-11e5-8ed6-7196a864851d.png)

Ok! This fix will be in the upcoming 4.0 release. 
Going to see about getting it into 3.6 soon as well.

**Comment by mezz** [comment]
Submitted Forestry 3.6.7 to Curse with this fix.

**Comment by pyure** [comment]
mezz, you're a hero.  Thanks again for putting so much time into annoying cross-mod compatibility issues.
